### PR TITLE
feat(arm): PostgressSQLGeoBackupEnabled

### PR DESCRIPTION
**Pull request by esterKoren**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*We converted the check PostgressSQLGeoBackupEnabled from TERRAFORM language to the ARM language so that it also works on resources that are defined in the ARM language.*

Fixes # (issue)

### Description
*Ensure that PostgreSQL server enables geo-redundant backups.*

### Fix
*o ensure that the PostgreSQL server enables geo-redundant backups in Azure ARM templates and validate it using Checkov, ensure the PostgresSQLGeoBackupEnabled check inspects properties.storageProfile.geoRedundantBackup for "enabled" and update the test case to verify against correctly structured ARM templates for pass and fail scenarios.*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/arm/checks/resource/example_PostgressSQLGeoBackupEnabled/fail1.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      "name": "fail1",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "administratorLogin": "psqladminun",
+        "administratorLoginPassword": "H@Sh1CoR3!",
+        "version": "9.6",
+        "storageProfile": {
+          "storageMB": 640000,
+          "backupRetentionDays": 7,
+          "geoRedundantBackup": "disabled"
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_PostgressSQLGeoBackupEnabled/fail1.json:17]: Disable

**Review by ChanochShayner — APPROVED** [review]
LGTM

File: tests/arm/checks/resource/example_PostgressSQLGeoBackupEnabled/fail1.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "type": "Microsoft.DBforPostgreSQL/servers",
+      "apiVersion": "2017-12-01",
+      "name": "fail1",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "administratorLogin": "psqladminun",
+        "administratorLoginPassword": "H@Sh1CoR3!",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_PostgressSQLGeoBackupEnabled/fail1.json:12]:
```suggestion
        "administratorLoginPassword": "blabla",
```

File: tests/arm/checks/resource/example_PostgressSQLGeoBackupEnabled/fail2.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "type": "Microsoft.DBforPostgreSQL/servers",
+      "apiVersion": "2017-12-01",
+      "name": "fail2",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "administratorLogin": "psqladminun",
+        "administratorLoginPassword": "H@Sh1CoR3!",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_PostgressSQLGeoBackupEnabled/fail2.json:12]:
```suggestion
        "administratorLoginPassword": "blabla",
```

File: tests/arm/checks/resource/example_PostgressSQLGeoBackupEnabled/pass.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "type": "Microsoft.DBforPostgreSQL/servers",
+      "apiVersion": "2017-12-01",
+      "name": "pass",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "administratorLogin": "psqladminun",
+        "administratorLoginPassword": "H@Sh1CoR3!",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_PostgressSQLGeoBackupEnabled/pass.json:12]:
```suggestion
        "administratorLoginPassword": "blabla",
```

**Review by tsmithv11 — APPROVED** [review]

**Pull request merged**